Dev1

I1kham opened the pull request:
9 спринт

I1kham committed:
8 сприн

I1kham committed:
8 спринт

I1kham committed:
8 спринт
2ая попытка

I1kham committed:
8 спринт
2ая попытка
доработка

I1kham committed:
8 спринт
2ая попытка, снова

I1kham committed:
9 спринт

faritowich commented on app/src/main/java/com/alchemtech/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,61 @@
+            }
+            val shareIntent =
+                Intent.createChooser(buttonShareAppIntent, getString(R.string.buttonShareAppTitle))
+            startActivity(shareIntent)
```
Можно упростить до 
```
Intent().apply {
    action = Intent.ACTION_SEND
    putExtra(Intent.EXTRA_TEXT, getString(R.string.buttonShareApp))
    type = "text/plain"
    startActivity(Intent.createChooser(this, getString(R.string.buttonShareAppTitle)))
}             
```

faritowich commented on app/src/main/java/com/alchemtech/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,79 @@
+    return if (s.isNullOrEmpty()) {
+        View.GONE
+    } else {
+        View.VISIBLE
```
Для изменения видимости элемента на экране в данном случае удобнее использовать extension View.isVisible:
View.isVisible = true соответствует View.visibility = View.VISIBLE
View.isVisible = false соответствует View.visibility = View.GONE

faritowich commented on app/src/main/java/com/alchemtech/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,79 @@
+        const val STRING_AMOUNT = "STRING_AMOUNT"
+        const val AMOUNT_DEF = ""
+
+    }
```
companion object принято объявлять в конце класса (реже в начале), но не посреди других методов класса

faritowich commented on app/src/main/java/com/alchemtech/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,79 @@
+// Ключ должен быть константным, чтобы мы точно знали, что он не изменится
+    companion object {
+        const val STRING_AMOUNT = "STRING_AMOUNT"
+        const val AMOUNT_DEF = ""
```
private

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,41 @@
+    <style name="settingsActivityButtonsStyle" parent="@style/Widget.AppCompat.Button.Borderless">
+        <item name="android:theme">@style/Theme.SettingsActivity</item>
+        <item name="android:layout_height">wrap_content</item>
+        <item name="android:layout_marginHorizontal">16dp</item>
```
Лучше внешние отступы, высоту и ширину, веса, гравити не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи и вообще только за то, что находится внутри нее - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

I1kham committed:
9 спринт
заметка для себя

I1kham committed:
9 спринт
переработан ui, измененна логика кнопки очистки текста, глубже залез в настройку ночной и дневной темы

faritowich commented on app/src/main/java/com/alchemtech/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,74 @@
+                if (savedInstanceState != null) {
+                    onSaveInstanceState(savedInstanceState)
+                    inputString= s.toString()
+                }
```
Всё еще сделано неправильно. 
1) onSaveInstanceState должен вызываться только при смене конфигурации (перевороте устройства), а сейчас этот метод вызывается при вводе каждого символа. Надо в TextWatcher не вызывать savedInstanceState, а просто записывать новое значение s в inputString. 
2) затем в onCreate или в onRestoreInstanceState сеттить в inputEditText сохраненный текст

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -0,0 +1,73 @@
+            android:layout_marginEnd="4dp"
+            android:layout_gravity="center"
+            android:background="@color/transparent"
+            android:src="@android:drawable/ic_menu_close_clear_cancel"
```
1) Иконка должна быть векторной
2) Она должна быть в drawable

I1kham committed:
9 спринт
переработан ui, измененна логика кнопки очистки текста, глубже залез в настройку ночной и дневной темы
